### PR TITLE
Add upgrade guide to version 2.0

### PR DESCRIPTION
**Pull request by OndraM**
I believe this would help API Client users a lot and encourage them for upgrade.

But I'm not quite sure all the examples are valid, so @foglcz please review :).

Also any feedback from others is welcome.

### CODE REVIEW

File: UPGRADE-2.0.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+```php
+$recommendation = UserRecommendation::create('user-id', 5, 'scenario', 1.0, 3600);
+```
+
+#### After
+
+```php
+$recommendation = UserRecommendation::create('user-id', 5, 'scenario', 1.0, 3600);
+$recommendation->addFilter('valid_to >= NOW()');
```
Comment (OndraM) [inline UPGRADE-2.0.md:79]: Is this true? Or what is 1:1 equivalent of the old behavior - how should the filter look now?

@foglcz ?

File: UPGRADE-2.0.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+$recommendation = UserRecommendation::create('user-id', 5, 'scenario', 1.0, 3600);
+$recommendation->addFilter('foo = bar')
+    ->addFilter('item_id not in {"123456789"}');
+```
+
+#### After
+
+```php
+$recommendation = UserRecommendation::create('user-id', 5, 'scenario', 1.0, 3600);
+$recommendation->addFilter("foo = 'bar'")
```
Comment (OndraM) [inline UPGRADE-2.0.md:98]: I'm not sure the filters are properly rewritten to MQL.

@foglcz could you please check it and maybe also provide some other 1:1 examples of some current common filters being used by Matej customers?

BTW do you really need to encapsulate filed contents using single quotes `'`, eg. `addFilter("foo = 'bar'")`? Or double quotes `"` works as well - eg. `addFilter('foo = "bar"')`?

File: UPGRADE-2.0.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+// Additional properties could now be passed to the UserRecommendation command using constructor or setResponseProperties() method
+$recommendation = UserRecommendation::create('user-id', 5, 'scenario', 1.0, 3600, ['item_title']);
+
+$response = $matej->request()
+    ->recommendation($recommendation)
+    ->send();
+
+$recommendations = $response->getRecommendation()->getData();
+
+// $recommendations now contains array of stdClasse with item_id and other requested properties (item_title in this example)
```
Comment (jirinovak) [inline UPGRADE-2.0.md]: typo `stdClasses`

**Review by jirinovak — COMMENTED** [review]
Thanks, it would be helpful.

**Review by foglcz — APPROVED** [review]
Nice, thanks! :)

**Comment by ghost** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Backward incompatible changes detected
  </td>
    </tr>
  </tbody>
</table>




## Backward incompatible changes introduced by this PR
 - [BC] The return type of Lmc\Matej\Model\Command\UserForget#getForgetMethod() changed from string to the non-covariant Lmc\Matej\Model\Command\Constants\UserForgetMethod
 - [BC] The return type of Lmc\Matej\Model\Command\UserForget#getForgetMethod() changed from string to Lmc\Matej\Model\Command\Constants\UserForgetMethod
 - [BC] The parameter $minimalRelevance of Lmc\Matej\Model\Command\UserRecommendation#setMinimalRelevance() changed from string to a non-contravariant Lmc\Matej\Model\Command\Constants\MinimalRelevance
 - [BC] The parameter $minimalRelevance of Lmc\Matej\Model\Command\UserRecommendation#setMinimalRelevance() changed from string to Lmc\Matej\Model\Command\Constants\MinimalRelevance
 - [BC] Constant Lmc\Matej\Model\Command\ItemPropertySetup::PROPERTY_TYPE_INT was removed
 - [BC] Constant Lmc\Matej\Model\Command\ItemPropertySetup::PROPERTY_TYPE_DOUBLE was removed
 - [BC] Constant Lmc\Matej\Model\Command\ItemPropertySetup::PROPERTY_TYPE_STRING was removed
 - [BC] Constant Lmc\Matej\Model\Command\ItemPropertySetup::PROPERTY_TYPE_BOOLEAN was removed
 - [BC] Constant Lmc\Matej\Model\Command\ItemPropertySetup::PROPERTY_TYPE_TIMESTAMP was removed
 - [BC] Constant Lmc\Matej\Model\Command\ItemPropertySetup::PROPERTY_TYPE_SET was removed
 - [BC] Constant Lmc\Matej\Model\Command\UserForget::ANONYMIZE was removed
 - [BC] Constant Lmc\Matej\Model\Command\UserForget::DELETE was removed
 - [BC] Method Lmc\Matej\Model\Command\UserForget#setForgetMethod() was removed
 - [BC] Constant Lmc\Matej\Model\Command\UserRecommendation::MINIMAL_RELEVANCE_LOW was removed
 - [BC] Constant Lmc\Matej\Model\Command\UserRecommendation::MINIMAL_RELEVANCE_MEDIUM was removed
 - [BC] Constant Lmc\Matej\Model\Command\UserRecommendation::MINIMAL_RELEVANCE_HIGH was removed


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**